ui/src/app/base/components/Header/Header.js

```diff
@@ -0,0 +1,220 @@
+import { Link } from "react-router-dom";
+import { withRouter } from "react-router";
+import classNames from "classnames";
+import React, { useState } from "react";
+import ReactRouterPropTypes from "react-router-prop-types";
+
+import "./Header.scss";
+import { AuthUserShape } from "../../proptypes";
+
+export const Header = ({ authUser, location }) => {
+  const [hardwareVisible, toggleHardware] = useState(false);
+  const [mobileMenuVisible, toggleMobileMenu] = useState(false);
+
+  const _toggleMobileMenu = evt => {
```

Comment:
These toggles and event handlers could be cleaned up nicely with a `useToggle` hook. David Ceddia has written a post about exactly that here: https://daveceddia.com/custom-hooks/ 

What do you think?